src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -507,8 +507,13 @@ internal void WriteCore(ReadOnlySpan<byte> buffer)
             EnsureCompressionMode();
             EnsureNotDisposed();
 
-            Debug.Assert(_deflater != null);
+            if (buffer.IsEmpty)
+            {
+                return;
+            }
+
             // Write compressed the bytes we already passed to the deflater:
```

Comment:
I know it's not from your change, but shouldn't this message say this?:

```suggestion
            // Write the compressed bytes we already passed to the deflater:
```

Reply:
Technically I think it's more correct as it currently is, even though it reads awkwardly.  We didn't pass compressed bytes: we passed uncompressed bytes, and we're then writing out the result, so we're writing, compressed, the bytes we already passed to the defaltor.

Reply:
Oh! I see it now. it's weird but now it makes sense.